### PR TITLE
Transcoder: can extract several channels from an input stream, and encode them to one output stream

### LINKED ISSUE #151: Transcoder : do not set output profile AND ICodec when adding generated stream
<a href="https://github.com/valnoel"></img></a> **Issue by [valnoel](https://github.com/valnoel)**
_Monday Jan 19, 2015 at 18:31 GMT_
_Originally opened as https://github.com/mikrosimage/avTranscoder/issues/60_

----

When we want to generate a stream (black or silence), we should not have to set both output profile and ICodec. The ICodec should be guessed from the profile, which must contain enough parameters to generate the stream.
It would avoid duplicated parameters into ICodec and profile.

https://github.com/mikrosimage/avTranscoder/blob/develop/src/AvTranscoder/transcoder/Transcoder.hpp#L56
https://github.com/mikrosimage/avTranscoder/blob/develop/src/AvTranscoder/transcoder/Transcoder.hpp#L66

### LINKED ISSUE #265: Transcoder: be able to wrap data to multi-channels streams
Update StreamTranscoder class.

### PR DESCRIPTION
**Pull request by cchampet**
Fix #265
Fix #151

### CODE REVIEW

File: src/AvTranscoder/transcoder/Transcoder.hpp [hunk 1]
```diff
@@ -163,7 +163,7 @@ class AvExport Transcoder
     void addTranscodeStream(const InputStreamDesc& inputStreamDesc,
                             const ProfileLoader::Profile& profile, const float offset = 0);
 
-    void addDummyStream(const ProfileLoader::Profile& profile);
+    void addGeneratedStream(const ProfileLoader::Profile& profile);
```
Comment (valnoel) [inline src/AvTranscoder/transcoder/Transcoder.hpp]: Sorry but once more, I am discussing the name of a method: here at first reading, it seems _addGeneratedStream_ means you "add a stream that has already been generated", while you would like to "add a stream that will be generated by the transcoder". So I'd prefer something like _addStreamGeneration_,  _addStreamToGenerated_ or _addStreamGenerator_, which seems clearer to me. Don't you think?

File: test/pyTest/testTranscoderTranscodeAudioMov.py [hunk 2]
```diff
@@ -52,7 +52,7 @@ def testTranscodeMovVariableNbSamplesPerFrame():
 def testTranscodeMovExtractChannels():
     """
     Transcode the audio stream of a MOV file which contains a video stream.
-    Extract channel one and third of the audio stream (5.1).
+    Extract channel one and third of the audio stream (5.1), and create two output streams.
```
Comment (valnoel) [inline test/pyTest/testTranscoderTranscodeAudioMov.py]: According to 
```c++
transcoder.addStream( av.InputStreamDesc(inputFileName, audioStreamIndex, 0) )
transcoder.addStream( av.InputStreamDesc(inputFileName, audioStreamIndex, 3) )
```
it seems it should be: "Extract channel one and __four__ of the audio stream (5.1)..."

File: src/AvTranscoder/transcoder/Transcoder.hpp (hunk 1 above)
Comment (cchampet) [inline src/AvTranscoder/transcoder/Transcoder.hpp]: Result of the vote: "addGenerateStream".

Comment (cchampet) [inline src/AvTranscoder/transcoder/Transcoder.hpp]: See commit 584f530

File: test/pyTest/testTranscoderTranscodeAudioMov.py (hunk 2 above)
Comment (cchampet) [inline test/pyTest/testTranscoderTranscodeAudioMov.py]: See commit d01ccc8

**Pull request merged**